device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -357,99 +328,38 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
                                                in_params_buffer);
         TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
 
-        /*****************************************************************
-         * Kernel3: Count the number of measurements per parameter
-         ****************************************************************/
-
-        vecmem::data::vector_buffer<unsigned int> n_measurements_buffer(
-            n_in_params, m_mr.main);
-        vecmem::device_vector<unsigned int> n_measurements_device(
-            n_measurements_buffer);
-        thrust::fill(thrust::cuda::par.on(stream),
-                     n_measurements_device.begin(), n_measurements_device.end(),
-                     0u);
-
-        // Create a buffer for the first measurement index of parameter
-        vecmem::data::vector_buffer<unsigned int> ref_meas_idx_buffer(
-            n_in_params, m_mr.main);
-
-        nThreads = m_warp_size * 2;
-        nBlocks = (n_in_params + nThreads - 1) / nThreads;
-        kernels::count_measurements<<<nBlocks, nThreads, 0, stream>>>(
-            in_params_buffer, barcodes_buffer, upper_bounds_buffer, n_in_params,
-            n_measurements_buffer, ref_meas_idx_buffer,
-            (*global_counter_device).n_measurements_sum);
-        TRACCC_CUDA_ERROR_CHECK(cudaGetLastError());
-
-        // Global counter object: Device -> Host
-        TRACCC_CUDA_ERROR_CHECK(
-            cudaMemcpyAsync(&global_counter_host, global_counter_device.get(),
-                            sizeof(device::finding_global_counter),
-                            cudaMemcpyDeviceToHost, stream));
-
         m_stream.synchronize();
```

Comment:
Is this synchronization point needed? I think it was only here for the `cudaMemcpyAsync(...)` call, which is now removed. 🤔

Reply:
Indeed it is not. Removed.